bpf/lib/conntrack.h

```diff
@@ -369,6 +396,31 @@ ipv6_ct_tuple_swap_ports(struct ipv6_ct_tuple *tuple)
 	tuple->dport = tmp;
 }
 
+static __always_inline int
+ct_extract_ports6(struct __ctx_buff *ctx, int off,
+		  enum ct_dir dir __maybe_unused, struct ipv6_ct_tuple *tuple,
+		  bool *has_l4_header __maybe_unused)
+{
+	switch (tuple->nexthdr) {
+	case IPPROTO_TCP:
+	case IPPROTO_UDP:
+		/* Load sport + dport into tuple. */
+		if (ctx_load_bytes(ctx, off, &tuple->dport, 4) < 0)
+			return DROP_CT_INVALID_HDR;
+
+		break;
+
+	default:
+		/* Can't handle ICMP or extension headers yet. */
```

Comment:
What does it mean in practice if we don't have ICMPv6 support here? Looks like we wouldn't be able to classify traffic as belonging to an inbound connection, and miss to skip SNAT?

Reply:
Hmm good question, yes possibly. I added commit https://github.com/cilium/cilium/pull/23165/commits/410aac7ec149a711ae43b954fa837328c79e6853 to refactor a bit based on what Maxim did for IPv4, and to add ICMP support here as well. Can you please take a look?

Reply:
thanks, will stare at it! @gentoo-root FYI

Reply:
looks good, ty!